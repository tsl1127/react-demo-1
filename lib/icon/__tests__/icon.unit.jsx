import renderer from 'react-test-renderer'
import React from 'react'
import Icon from '../icon'
import {mount} from 'enzyme'

describe('icon测试',()=>{
    it('是个svg',()=>{
        const json = renderer.create(<Icon name="wechat"/>).toJSON()  //变为json
        expect(json).toMatchSnapshot()  //match快照,如果需要更新快照则npm test -- -u
    })
    it('测试onclick',()=>{
        let n = 1
        const fn = ()=>{
            n = 2
        }
        const c = mount(<Icon name="wechat" onClick={fn}></Icon>)
        c.find('svg').simulate('click')
        expect(n).toEqual(2)
    })
})