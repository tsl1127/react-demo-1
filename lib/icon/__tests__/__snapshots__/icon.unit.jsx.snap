// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon测试 是个svg 1`] = `
<svg
  className="lunzi-icon"
>
  <use
    xlinkHref="#wechat"
  />
</svg>
`;
